Failure # 1 (occurred at 2022-03-06_18-26-14)
Traceback (most recent call last):
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/tune/trial_runner.py", line 893, in _process_trial
    results = self.trial_executor.fetch_result(trial)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/tune/ray_trial_executor.py", line 707, in fetch_result
    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/worker.py", line 1733, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(TuneError): [36mray::ImplicitFunc.train()[39m (pid=8294, ip=192.168.100.15, repr=trainable)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/tune/trainable.py", line 315, in train
    result = self.step()
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/tune/function_runner.py", line 381, in step
    self._report_thread_runner_error(block=True)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/tune/function_runner.py", line 531, in _report_thread_runner_error
    raise TuneError(
ray.tune.error.TuneError: Trial raised an exception. Traceback:
[36mray::ImplicitFunc.train()[39m (pid=8294, ip=192.168.100.15, repr=trainable)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/tune/function_runner.py", line 262, in run
    self._entrypoint()
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/tune/function_runner.py", line 330, in entrypoint
    return self._trainable_func(self.config, self._status_reporter,
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/ray/tune/function_runner.py", line 600, in _trainable_func
    output = fn()
  File "training.py", line 85, in trainable
    model = model.fit(X=all_features, y=all_labels)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/classifier.py", line 142, in fit
    return super(NeuralNetClassifier, self).fit(X, y, **fit_params)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/net.py", line 917, in fit
    self.partial_fit(X, y, **fit_params)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/net.py", line 876, in partial_fit
    self.fit_loop(X, y, **fit_params)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/net.py", line 789, in fit_loop
    self.run_single_epoch(dataset_train, training=True, prefix="train",
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/net.py", line 826, in run_single_epoch
    step = step_fn(Xi, yi, **fit_params)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/net.py", line 723, in train_step
    self.optimizer_.step(step_fn)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/optim/optimizer.py", line 89, in wrapper
    return func(*args, **kwargs)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/optim/sgd.py", line 87, in step
    loss = closure()
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/net.py", line 719, in step_fn
    step = self.train_step_single(Xi, yi, **fit_params)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/net.py", line 659, in train_step_single
    y_pred = self.infer(Xi, **fit_params)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/skorch/net.py", line 1062, in infer
    return self.module_(x, **fit_params)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/uriel/Documentos/CIC/DCC/Papers/ANN as Feature Selection/attn-fs-archs/src/ndsl/architecture/attention.py", line 205, in forward
    output = self.transformer_encoder(embeddings).transpose(0, 1)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/uriel/Documentos/CIC/DCC/Papers/ANN as Feature Selection/attn-fs-archs/src/ndsl/architecture/attention.py", line 90, in forward
    output, weights = mod(output, src_mask=mask)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/uriel/Documentos/CIC/DCC/Papers/ANN as Feature Selection/attn-fs-archs/src/ndsl/architecture/attention.py", line 45, in forward
    src2, weights = self.self_attn(src, src, src, attn_mask=src_mask)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torchtext/nn/modules/multiheadattention.py", line 102, in forward
    attn_output, attn_output_weights = self.attention_layer(q, k, v, attn_mask=attn_mask,
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torchtext/nn/modules/multiheadattention.py", line 208, in forward
    attn_output_weights = torch.nn.functional.dropout(attn_output_weights, p=self.dropout, training=self.training)
  File "/home/uriel/Miniconda3/envs/DCC-attn/lib/python3.8/site-packages/torch/nn/functional.py", line 1076, in dropout
    return _VF.dropout_(input, p, training) if inplace else _VF.dropout(input, p, training)
RuntimeError: CUDA out of memory. Tried to allocate 162.00 MiB (GPU 0; 5.79 GiB total capacity; 4.34 GiB already allocated; 106.50 MiB free; 4.35 GiB reserved in total by PyTorch)

